src/EditorFeatures/CSharpTest/Completion/CompletionProviders/KeywordCompletionProviderTests.cs

```diff
@@ -408,5 +408,134 @@ public async Task SuggestEventAfterReadonlyInStruct()
 ";
             await VerifyItemExistsAsync(markup, "event");
         }
+
+        [Theory, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [WorkItem(39265, "https://github.com/dotnet/roslyn/issues/39265")]
+        [InlineData("struct", true)]
+        [InlineData("class", false)]
+        [InlineData("interface", false)]
+        public async Task SuggestReadonlyPropertyAccessor(string declarationType, bool present)
+        {
+
+            var markup =
+$@"{declarationType} C {{
+    int X {{
+        $$
+    }}
+}}
+";
```

Comment:
I would Honestly just prefer separate tests. I don't this form of test nearly impossible to read.

Reply:
I heavily prefer the use of a `Theory`, specifically for the `declarationType` parameter. When the test is broken out into separate tests, it is no longer possible/practical to see two things:

1. Are all cases covered? If it's not a `Theory`, then assume not.
2. Are all related cases covering the same thing? If a typo is fixed in one test input, was it fixed in all? If it's not a `Theory`, then one or more tests is probably wrong.